lighthouse-core/computed/critical-request-chains.js

```diff
@@ -66,6 +67,53 @@ class CriticalRequestChains {
     return ['VeryHigh', 'High', 'Medium'].includes(request.priority);
   }
 
+  /**
+   * Create a tree of critical requests.
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {LH.Gatherer.Simulation.GraphNode} graph
+   * @return {LH.Artifacts.CriticalRequestNode}
+   */
+  static extractChainUsingLantern(mainResource, graph) {
+    /** @type {LH.Artifacts.CriticalRequestNode} */
+    const rootNode = {};
+
+    /**
+     * @param {LH.Gatherer.Simulation.GraphNetworkNode[]} path
+     */
+    function addChain(path) {
+      let currentNode = rootNode;
+
+      for (const node of path) {
+        if (!currentNode[node.id]) {
+          currentNode[node.id] = {
+            request: node.record,
+            children: {},
+          };
+        }
+
+        currentNode = currentNode[node.id].children;
+      }
+    }
+
+    graph.traverse((node, traversalPath) => {
+      if (node.type !== 'network') return;
+      if (!CriticalRequestChains.isCritical(node.record, mainResource)) return;
+
+      const networkPath = traversalPath
+        .filter(/** @return {n is LH.Gatherer.Simulation.GraphNetworkNode} */
+          n => n.type === 'network')
+        .reverse();
+
+      // Ignore if some ancestor is not a critical request. Ignores root node.
```

Comment:
why ignore the root node? shouldn't it always be critical?

Reply:
I get an empty chain for `CriticalRequestChain gatherer: extractChain function › handles non-existent nodes when building the tree` when we don't skip the first...